snuba/datasets/profiles_processor.py

```diff
@@ -9,44 +9,80 @@
+                additional_fields = {
+                    f"device_{field}": message.get(f"device_{field}", "")
+                    for field in ["locale", "classification", "manufacturer", "model"]
+                }
+                additional_fields["architecture"] = message.get(
+                    "architecture", "unknown"
+                )
+                additional_fields["version_name"] = message["version_name"]
+                additional_fields["device_os_name"] = message["device_os_name"]
+                additional_fields["device_os_version"] = message["device_os_version"]
+                additional_fields["duration_ns"] = message["duration_ns"]
+                additional_fields["transaction_name"] = message["transaction_name"]
+                additional_fields["transaction_id"] = str(
+                    UUID(message["transaction_id"])
+                )
+                additional_fields["trace_id"] = str(UUID(message["trace_id"]))
+                additional_fields["profile_id"] = str(UUID(message["profile_id"]))
+
             processed = {
-                "organization_id": message["organization_id"],
-                "project_id": message["project_id"],
-                "transaction_id": str(UUID(message["transaction_id"])),
-                "profile_id": str(UUID(message["profile_id"])),
-                "received": datetime.utcfromtimestamp(message["received"]),
-                "profile": message["profile"],
                 "android_api_level": message.get("android_api_level"),
-                "device_classification": message["device_classification"],
-                "device_locale": message["device_locale"],
-                "device_manufacturer": message["device_manufacturer"],
-                "device_model": message["device_model"],
                 "device_os_build_number": message.get("device_os_build_number"),
-                "device_os_name": message["device_os_name"],
-                "device_os_version": message["device_os_version"],
-                "architecture": message.get("architecture", "unknown"),
-                "duration_ns": message["duration_ns"],
                 "environment": message.get("environment"),
-                "platform": message["platform"],
-                "trace_id": str(UUID(message["trace_id"])),
-                "transaction_name": message["transaction_name"],
-                "version_name": message["version_name"],
-                "version_code": message["version_code"],
-                "retention_days": retention_days,
                 "offset": metadata.offset,
+                "organization_id": message["organization_id"],
                 "partition": metadata.partition,
+                "platform": message["platform"],
+                "profile": "",  # deprecated
+                "project_id": message["project_id"],
+                "received": datetime.utcfromtimestamp(message["received"]),
+                "retention_days": message["retention_days"],
```

Comment:
is there a reason that the retention days aren't enforced to be `30` or `90` anymore?

Reply:
We pull the value from https://github.com/getsentry/sentry/blob/master/src/sentry/profiles/task.py#L148 now and it didn't seem useful to have some guard here to switch between 30 or 90, purely because we might end up with a different value at some point (like 60). Before, we were kind of setting it ourselves and it was definitely a good idea to guard it.